### ISSUE TITLE
Support for obsidian style highlighting (of the form -> ==<text>==)

### ISSUE THREAD

**Issue by cdmill** [epwalsh/obsidian.nvim#241]
### 🚀 The feature, motivation and pitch

I am wondering if any plans have been made to include the highlighting syntax (delimited by ==, eg ==text==) which is supported in obsidian itself. 

I currently do this manually:

```vim
" ~/.config/nvim/after/syntax/markdown.vim

syn region highlight_inline matchgroup=texStatement start="==" end="==" concealends
hi highlight_inline guibg=#6a491d
```

But I think this could be a nice ui addition. As of right now this type of highlighting isn't yet included in CommonMark, as such it isn't available in the markdown highlighting provided by plugins like treesitter (as far as I know). 

### Additional context

Here is an example of how it looks on my machine:

<img width="541" alt="sc 2023-11-21 at 14 12 20" src="https://github.com/epwalsh/obsidian.nvim/assets/115658917/5c7be125-c9af-41ce-9c92-3fc9d3a6d731">

**Comment by epwalsh** [comment]
Hey @cdmill this looks like it would be fairly straightforward to add. In case someone beats me to it, here's what you should do:

1. Add a function to `lua/obsidian/ui.lua` called `get_line_highlight_extmarks` with the same API as [`get_line_check_extmarks()`](https://github.com/epwalsh/obsidian.nvim/blob/c2c5d69e1d69857b7554aedc67d1320e9cd42a87/lua/obsidian/ui.lua#L197). This should add two extmarks to conceal the start and end "==", and then another extmark to highlight the text in-between with a new highlight group (the HL group needs to be added to `config.ui`, see step 3 below). For an example of how to conceal and highlight, see [this block](https://github.com/epwalsh/obsidian.nvim/blob/c2c5d69e1d69857b7554aedc67d1320e9cd42a87/lua/obsidian/ui.lua#L228-L264).
2. Make sure that function is called alongside the others [here](https://github.com/epwalsh/obsidian.nvim/blob/c2c5d69e1d69857b7554aedc67d1320e9cd42a87/lua/obsidian/ui.lua#L414-L415).
3. Add a new default highlight group to the config [here](https://github.com/epwalsh/obsidian.nvim/blob/c2c5d69e1d69857b7554aedc67d1320e9cd42a87/lua/obsidian/config.lua#L238) and a new option called `highlight_text` or something like that, which should look like the [`reference_text`](https://github.com/epwalsh/obsidian.nvim/blob/c2c5d69e1d69857b7554aedc67d1320e9cd42a87/lua/obsidian/config.lua#L236) option.

**Comment by cdmill** [comment]
Sounds good, I'll submit a PR once done and let you know if any issues arise 👍

**Comment by epwalsh** [comment]
Great. FYI I just pushed some widespread, internal organization changes so please pull those in before you get too deep, otherwise you might end up with a bunch of merge conflicts.

**Comment by epwalsh** [comment]
Implemented in #242
